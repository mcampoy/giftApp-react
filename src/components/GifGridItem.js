import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ( {title, url} ) => {

    return (
        <div className="card animate__animated animate__fadeIn ">
            <img src={url} alt={title} loading="lazy" />
            {/* <p>{title}</p> */}
        </div>
    )
   
}

    GifGridItem.propTypes = {
        url: PropTypes.string.isRequired
    }

export default GifGridItem
