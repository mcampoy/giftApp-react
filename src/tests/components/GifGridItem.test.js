import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Se prueban el componente GifGridItem', () => {

    const title = 'Soy un párrafo';
    const url = 'https//urldeprueba.jpg';
    let wrapper = shallow( < GifGridItem  title={ title } url={url}/> );

    test('Debe mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    });

    // test('debe tener un párrafo con el título', () => {

    //     const p = wrapper.find('p');
    //     expect( p.text().trim() ).toBe( title );

    // });

    test('debe tener la imagen igual al url y alt de las props', () => {

        const img = wrapper.find('img');
        // console.log(img.props().src)
        // console.log(img.prop('src'))

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    })

    test('debe tener la clase animate_fadeIn', () => {

        const div = wrapper.find('div');

        // expect(div.hasClass('animate__fadeIn')).toBeTruthy();

        const className = div.prop('className');

        expect( className.includes('animate__fadeIn')).toBe(true);


    })
    



})