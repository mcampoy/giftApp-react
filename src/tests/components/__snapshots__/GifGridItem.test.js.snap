// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Se prueban el componente GifGridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn "
>
  <img
    alt="Soy un pÃ¡rrafo"
    loading="lazy"
    src="https//urldeprueba.jpg"
  />
</div>
`;
